lightning-background-processor/src/lib.rs

```diff
@@ -236,26 +236,28 @@ fn handle_network_graph_update<L: Deref>(
 }
 
 fn update_scorer<'a, S: 'static + Deref<Target = SC> + Send + Sync, SC: 'a + WriteableScore<'a>>(
-	scorer: &'a S, event: &Event
+	scorer: &'a S,
+	event: &Event,
+	score_params: &<S::Target as Score>::ScoreParams
 ) {
 	let mut score = scorer.lock();
 	match event {
 		Event::PaymentPathFailed { ref path, short_channel_id: Some(scid), .. } => {
-			score.payment_path_failed(path, *scid);
+			score.payment_path_failed(path, *scid, score_params);
```

Comment:
>Thus, I think we shouldn't change the update methods on the score but rather only update the fetch-score method.

this means to just take out the scoreparams from 
`fn payment_path_failed`
`fn payment_path_successful`
`fn probe_failed`
`fn probe_successful` 

and only apply in 
`fn channel_penalty_msat` 
because `channel_penalty_msat` is the "fetch-score" method

<img width="1000" alt="image" src="https://user-images.githubusercontent.com/906931/236006455-8a956883-9ec6-4c9f-9c05-6584fc8afac0.png">

Yup. i think that makes sense. i can make that change.